CHANGELOG.md

```diff
@@ -9,6 +9,34 @@
 * [CHANGE] Dropped "blank Alertmanager configuration; using fallback" message from Info to Debug level. #3205
 * [CHANGE] Zone-awareness replication for time-series now should be explicitly enabled in the distributor via the `-distributor.zone-awareness-enabled` CLI flag (or its respective YAML config option). Before, zone-aware replication was implicitly enabled if a zone was set on ingesters. #3200
 * [CHANGE] Removed the deprecated CLI flag `-config-yaml`. You should use `-schema-config-file` instead. #3225
+* [CHANGE] Enforced the HTTP method required by some API endpoints which did (incorrectly) allow any method before that. #3228
+  - `GET /`
+  - `GET /config`
+  - `GET /debug/fgprof`
+  - `GET /distributor/all_user_stats`
+  - `GET /distributor/ha_tracker`
+  - `GET /all_user_stats`
+  - `GET /ha-tracker`
+  - `GET /api/v1/user_stats`
+  - `GET /api/v1/chunks`
+  - `GET <legacy-http-prefix>/user_stats`
+  - `GET <legacy-http-prefix>/chunks`
+  - `GET /services`
+  - `GET /multitenant_alertmanager/status`
+  - `GET /status` (alertmanager microservice)
+  - `GET|POST /ingester/ring`
+  - `GET|POST /ring`
+  - `GET|POST /store-gateway/ring`
+  - `GET|POST /compactor/ring`
+  - `GET|POST /ingester/flush`
```

Comment:
> push POST-only

It is POST only but the CHANGELOG was wrong. Fixed, thanks!

> Perhaps we should make /flush, /shutdown

I agree would be safer. If do so, we should consider:
1. Change `tools/migrate-ingester-statefulsets.sh` because calls `GET /shutdown`, but then we would have to use something different than `wget` (and `curl` is not available in Cortex images)
2. We would have to remove these links from the `/` HTML page because if you click on that they will fail (so there's no point linking them, the https://cortexmetrics.io/docs/api/ will be enough)

Thoughts?

Reply:
> 1. Change `tools/migrate-ingester-statefulsets.sh` because calls `GET /shutdown`, but then we would have to use something different than `wget` (and `curl` is not available in Cortex images)

Valid point. What about we include little tool in the image to call those endpoints? We could fix another problems with included wget too -- it doesn't like 204. I'd suggest separate PR for all that.

> 2. We would have to remove these links from the `/` HTML page because if you click on that they will fail (so there's no point linking them, the https://cortexmetrics.io/docs/api/ will be enough)

I would keep them in the index page (it's useful to know that given Cortex supports it), but add a note that they are POST-only.

Reply:
> Valid point. What about we include little tool in the image to call those endpoints? We could fix another problems with included wget too -- it doesn't like 204. I'd suggest separate PR for all that.

I've the feeling it's overkilled. Installing `curl` in our Docker images could be just easier.

Reply:
`curl` also doesn't like 204 and returns non-zero exit code. We could simplify our script if we had this.

Reply:
+1 to making them `POST` only!
+1 to showing them in index page but mention that they only support `POST`.

Not too familiar with the script though.

Reply:
Would you agree if I do it in a separate PR? We need to find a solution for the `wget` replacement and I would like to avoid blocking this PR because of that.

Reply:
> Would you agree if I do it in a separate PR? We need to find a solution for the `wget` replacement and I would like to avoid blocking this PR because of that.

Absolutely

Reply:
Opened an issue for /flush and /shutdown:
https://github.com/cortexproject/cortex/issues/3243